sdk/python/kfp/dsl/_component.py

```diff
@@ -118,3 +120,44 @@ def _component(*args, **kargs):
     return container_op
 
   return _component
+
+def graph_component(func):
```

Comment:
Is it too hard to merge graph_component and component as one? e.g. if the graph has only single ContainerOp in the opsgroup and the return value is it, then it's a single container op. Otherwise, treat it as a graph component. This way we don't need to expose 2 interfaces here.

Reply:
It is a little tricky because the compiler can only differentiate the component from graph components when the function returns. However, the graph component creates a opsgroup in the Pipeline instance before the function is called. To make it possible, the compiler needs to undo the opsgroup insertion when it is not a graph component.

Reply:
Maybe we can keep the two decorators now and deprecate the graph_component once we have a full-fledged component decorator that does both of the two jobs. WDYT?
I'm ok with spending the efforts to merge both if you think it is critical to expose the two decorators as one. But I might need a bit of help on that.
Thanks

Reply:
I see. If it is not trivial, let's keep both for now. We can deprecate one of them in the future.

Reply:
Maybe add a TODO for tracking it.

Reply:
Sure.
My vision to combine both together is to have the decorator returns a base class that container_op and opsgroup inherit from because both are technically templates. Once we have such internal structures, it is trivial to combine both together.